core/services/im_here/TrafficController.py

```diff
@@ -0,0 +1,119 @@
+
+    def connection_made(self, transport: asyncio.BaseTransport) -> None:
+        """Behavior when a new connection is stablished."""
+        logger.debug(f"New UDP connection with {transport.get_extra_info('peername')}.")
+        self.transport = transport
+
+    def datagram_received(self, data: bytes, addr: Tuple[str, int]) -> None:
+        """What happens when data is received from a client socket."""
+        message = data.decode()
+        logger.info(f"Message received for component {self._component_id}: {message}")
+        mavlink_package = TrafficController.parse_mavlink_package(message)
+        TrafficController.forward_message(mavlink_package, self._component_id)
+        logger.info("Successfully forwarded mavlink coordinates package.")
+
+
+class TrafficController:
+    """Responsible for managing NMEA server sockets and traffic NMEA data between them and the Mavlink channel."""
+
+    def __init__(self) -> None:
+        self._socks: Dict[NMEASocketInfo, Union[asyncio.AbstractServer, asyncio.BaseTransport]] = {}
+
+    def get_socks(self) -> List[NMEASocketInfo]:
+        """Retrieve information about available server sockets."""
+        return list(self._socks)
+
+    async def add_sock(self, sock: NMEASocketInfo) -> None:
+        """Open a new network server socket and asyncronously wait for connections to arrive."""
+        loop = asyncio.get_running_loop()
+        server_socket: Union[asyncio.AbstractServer, asyncio.BaseTransport]
+        if sock.kind == SocketKind.TCP:
+            server_socket = await loop.create_server(lambda: TcpNmeaProtocol(sock.component_id), "0.0.0.0", sock.port)
+        elif sock.kind == SocketKind.UDP:
+            server_socket, _ = await loop.create_datagram_endpoint(
+                lambda: UdpNmeaProtocol(sock.component_id), local_addr=("0.0.0.0", sock.port)
+            )
+        else:
+            raise UnsupportedSocketKind(f"Got {sock.kind}. Expected one of: {[kind.value for kind in SocketKind]}.")
+        self._socks[sock] = server_socket
+        logger.debug(f"Added new sock: {sock}.")
+
+    def remove_sock(self, sock: NMEASocketInfo) -> None:
+        """Remove existing socket."""
+        server_socket = self._socks.pop(sock, None)
+        if server_socket is None:
+            raise ValueError(f"Socket {sock} does not exist.")
+        server_socket.close()
+        logger.debug(f"Removed sock. Socks now: {self.get_socks()}.")
+
+    @staticmethod
+    def parse_mavlink_package(nmea_msg: str) -> MavlinkGeoPackage:
+        """Transform NMEA message into proper Mavlink GPS_INPUT package."""
+        nmea_sentence = pynmea2.parse(nmea_msg)
+        return parse_mavlink_from_sentence(nmea_sentence)
+
+    @staticmethod
+    def forward_message(message: MavlinkGeoPackage, component_id: int) -> None:
+        """Forward Mavlink message package to Mavlink2Rest, on the specified component ID."""
+        mavlink2rest = MavlinkMessenger()
```

Comment:
This could use the build pattern

Reply:
You could do:
```py
mavlink2rest = MavlinkMessenger()
                            .set_component_id(component_id)
                            .send_mavlink_message(message.dict())
```

But that should not hold this PR

Reply:
Ohhhh ok. I was thinking about the TrafficController.

Indeed, that would be a cool change for MavlinkMessenger.